src/drivers/gitlab.js

```diff
@@ -181,18 +181,29 @@ class Gitlab {
 
     const endpoint = `/runners?per_page=100`;
     const runners = await this.request({ endpoint, method: 'GET' });
-    const runner = runners.filter(
+    const runner = runners.find(
       (runner) => runner.name === name || runner.description === name
-    )[0];
+    );
 
-    if (runner) return { id: runner.id, name: runner.name };
+    if (runner)
```

Comment:
This could be dryer.

I would do:

```js
async runnersByLabels(opts = {}) {
    const { labels } = opts;
    const endpoint = `/runners?per_page=100&${labels ? `tag_list=${labels}`: ''}`;
    const runners = await this.request({ endpoint, method: 'GET' });
    return runners.map((runner) => ({
      id: runner.id,
      name: runner.name,
      busy: runner.active,
      online: runner.status === 'online'
    }));
  }

async runnerByName(opts = {}) {
    const { name } = opts;
    const runners = await this.runnersByLabels();
    return runners.find(
      (runner) => runner.name === name || runner.description === name
    );
  }
```

RunnersbyName might be something very specific and a bit misleading. Feel free to do a runners method or as I did (to not break sync between drivers) we can assume that if you do not specify labels is returning all of them

Reply:
```
(runner) => runner.name === name || runner.description === name
```

should be

```
(runner) => runner.name === name 
```

Since description is not in the data model anymore

Reply:
> _RunnersbyName might be something very specific and a bit misleading_

Actually it's `runnerByName` without the plural, that's why I kept it: `runnersByLabels` returns runner**s** that contain all the specified labels, and `runnerByName` returns the **only** runner that has the specified unique name.

Reply:
You're going to laugh with this: actually, the `description` field is the only holding the name, and the `name` field is `null` in all the cases I could test.

Reply:
I've chosen to make `getRunners` universal on the driver and extract both `runnerByName` and `runnersByLabels` to `cml.js`

What do you think?

Reply:
Gitlab used description as far as I remember

Reply:
we treated description AS name